Userland/Utilities/file.cpp

```diff
@@ -163,66 +163,32 @@ static ErrorOr<Optional<String>> elf_details(StringView description, StringView
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/tinyvg"sv, "TinyVG vector graphics"sv, image_details)                        \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/webp"sv, "WebP image data"sv, image_details)                                 \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/x-portable-bitmap"sv, "PBM image data"sv, image_details)                     \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/x-portable-graymap"sv, "PGM image data"sv, image_details)                    \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/x-portable-pixmap"sv, "PPM image data"sv, image_details)                     \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/x-qoi"sv, "QOI image data"sv, image_details)                                 \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/x-targa"sv, "Targa image data"sv, image_details)                             \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("text/css"sv, "Cascading Style Sheet"sv, description_only)                          \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("text/csv"sv, "CSV text"sv, description_only)                                       \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("text/html"sv, "HTML document"sv, description_only)                                 \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("text/markdown"sv, "Markdown document"sv, description_only)                         \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("text/plain"sv, "plain text"sv, description_only)                                   \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("text/x-shellscript"sv, "POSIX shell script text executable"sv, description_only)   \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("video/webm"sv, "WebM video"sv, description_only)
+struct PatternAndFunction {
+    StringView matching_pattern;
+    ErrorOr<Optional<String>> (*details)(StringView, StringView);
+};
+
+static constexpr Array s_pattern_with_specialized_functions {
+    PatternAndFunction { "application/gzip"sv, gzip_details },
+    PatternAndFunction { "application/zip"sv, zip_details },
+    PatternAndFunction { "extra/elf"sv, elf_details },
+    PatternAndFunction { "audio/"sv, audio_details },
+    PatternAndFunction { "image/"sv, image_details },
+};
 
 static ErrorOr<Optional<String>> get_description_from_mime_type(StringView mime, StringView path)
 {
-#define __ENUMERATE_MIME_TYPE_DESCRIPTION(mime_type, description, details) \
-    if (mime_type == mime)                                                 \
-        return details(description, path);
-    ENUMERATE_MIME_TYPE_DESCRIPTIONS;
-#undef __ENUMERATE_MIME_TYPE_DESCRIPTION
-    return OptionalNone {};
+    auto const description = Core::get_description_from_mime_type(mime);
+
+    if (!description.has_value())
+        return OptionalNone {};
+
+    for (auto const& pattern : s_pattern_with_specialized_functions) {
+        if (mime.starts_with(pattern.matching_pattern))
```

Comment:
This feels like it might break in the future since not all patterns are prefixes...

Reply:
Absolutely, but I would rather not add useless (for now) branches. This can be simply solved by adding an `enum` to specialize how the pattern should be used, so no need to worry about it now.

Reply:
`StringView::matches()` allows the use of `*` as a wildcard. I think you can just change the image and audio patterns to "image/*" and "audio/*", and put `mime.matches(pattern.matching_pattern)` and it would Just Work :tm:.

Reply:
Sure, took me time to realize why you said "image/" and "audio/" without a `*` :smile: